calico/felix/dispatch.py

```diff
@@ -164,16 +165,36 @@ def _calculate_update(self, ifaces):
         # Special case: allow the metadata IP through from all interfaces.
         if self.config.METADATA_IP is not None and self.ip_version == 4:
             # Need to allow outgoing Metadata requests.
-            root_from_upds.append("--append %s "
-                                  "--protocol tcp "
-                                  "--in-interface %s+ "
-                                  "--destination %s "
-                                  "--dport %s "
-                                  "--jump RETURN" %
-                                  (CHAIN_FROM_ENDPOINT,
-                                   self.config.IFACE_PREFIX,
-                                   self.config.METADATA_IP,
-                                   self.config.METADATA_PORT))
+            root_inbound_upds.append("--append %s "
```

Comment:
The honest answer is that I took the path that made this look isomorphic to the rest of the work being done in Felix. It seemed like it didn't really cost anything, and had the advantage of providing a useful hook for extension.

If you want to take a different approach, that suits me.